[NFS] Upgrade chef recipe from `2.6.4` to `5.0.0`

EddyMM opened the pull request:
### Description of changes
* This commit upgrade the NFS Cookbook recipe version from `2.6.4` to `5.0.0`
* ChangeLog: [nfs/CHANGELOG.md](https://github.com/sous-chefs/nfs/blob/main/CHANGELOG.md)

### Tests
* Tested by creating an AMI based on the cookbook changes
* Created a cluster and checked the `chef-client` logs
* Successfully ran an MPI task
* Automated System Tests: Ubuntu & Centos7

### References
* [sous-chefs/nfs](https://github.com/sous-chefs/nfs)

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

lukeseawalker commented on cookbooks/aws-parallelcluster-config/recipes/nfs.rb:
```diff
@@ -22,10 +22,6 @@
-  # FIXME: https://github.com/atomic-penguin/cookbook-nfs/issues/93
-  include_recipe "nfs::server"
-end
 include_recipe "nfs::server4"
```
is this needed? Can this be moved in place of https://github.com/aws/aws-parallelcluster-cookbook/blob/develop/cookbooks/aws-parallelcluster-install/recipes/base.rb#L125 ?

lukeseawalker replied:
if we are calling `https://github.com/aws/aws-parallelcluster-cookbook/pull/1528` just to restart the service after the thread configuration, I'm for doing it "by hand" restarting the service, because calling the whole recipe may perform additional (not needed) actions

EddyMM replied:
The service restart seems to have been necessary for [Ubuntu1604](https://github.com/aws/aws-parallelcluster-cookbook/blob/12f9ecaae38afd9c4b505dccb15e3d10732bc4f4/cookbooks/aws-parallelcluster-config/recipes/nfs.rb#L19) to pick up the changes. It may no longer be needed in 18_04 or 20_04. I'll check on this.

lukeseawalker replied:
> The service restart seems to have been necessary ...

I think it's still needed for all the OS after setting 
`node.force_override['nfs']['threads'] = [node['cpu']['cores'].to_i, 8].max`
so that the service see the config change

EddyMM committed:
[NFS] Upgrade chef recipe from `2.6.4` to `5.0.0`

- ChangeLog: [nfs/CHANGELOG.md](https://github.com/sous-chefs/nfs/blob/main/CHANGELOG.md)

Signed-off-by: Eddy Mwiti <eddmwiti@amazon.com>
